Remove remnants of Repeat executor node.

hlinnaka opened the pull request:
It has been unused since the 9.5 merge, which replaced the way GROUPING
SETS are planned and executed with the upstream implementation.

pengzhout commented on src/backend/cdb/cdbplan.c:
```diff
@@ -211,17 +211,6 @@ plan_tree_mutator(Node *node,
 			}
 			break;
 
-		case T_Repeat:
```
T_Repeat seems to still be kept in node.h

pengzhout commented on src/include/executor/nodeRepeat.h:
```diff
@@ -1,25 +0,0 @@
-#include "nodes/execnodes.h"
-
-extern TupleTableSlot *ExecRepeat(RepeatState *repeatstate);
-extern RepeatState *ExecInitRepeat(Repeat *node, EState *estate, int eflags);
```
structure RepeatState should also be removed.

pengzhout replied:
Add T_RepeatState.

hlinnaka commented:
Thanks for the review @pengzhout, rebased and removed the stuff you pointed out. I'll push this after concourse has confirmed that everything is still green.
